src/pipeline_edsl/to_workflow.ml

```diff
@@ -284,6 +284,21 @@ module Make (Config : Compiler_configuration)
           ~name:(sprintf "Input file: %s" url)
       in
       Raw_file (deal_with_input_file raw_file (single_file ~host))
+    | Some "http" | Some "https" ->
+      let path =
+        let basename =
+          match Uri.get_query_param uri "filename" with
```

Comment:
Right-oh--mind commenting this? I like it, but it's not totally clear that this is for that, instead of for some weird other protocol.